mypy/server/astdiff.py

```diff
@@ -83,7 +83,10 @@ def is_similar_node_shallow(n: SymbolTableNode, m: SymbolTableNode) -> bool:
                 nn.is_newtype == mn.is_newtype and
                 is_same_mro(nn.mro, mn.mro))
     if isinstance(n.node, Var) and isinstance(m.node, Var):
-        return is_identical_type(n.node.type, m.node.type)
+        if n.node.type is None and m.node.type is None:
```

Comment:
Here the semantics is that if both nodes have ``None`` attributes, then ``is_similar_node_shallow`` returns ``True``. I am not this is right, so that I would like to hear from @JukkaL here (since he is working on fine grained incremetal).